### PR TITLE
Make datadog user-agent configurable

### PR DESCRIPTION
**Pull request by tiedotguy**
I'm a bit mixed on whether this is a 7.1.0 or an 8.0.0.  Can change it if people have opinions.

### CODE REVIEW

File: pkg/backends/datadog/datadog.go [hunk 1]
```diff
@@ -30,7 +30,7 @@ const (
 	// BackendName is the name of this backend.
 	BackendName                  = "datadog"
 	dogstatsdVersion             = "5.6.3"
-	dogstatsdUserAgent           = "python-requests/2.6.0 CPython/2.7.10"
+	defaultDogstatsdUserAgent    = "gostatsd"
```
Comment (aelse) [inline pkg/backends/datadog/datadog.go]: nit: defaultUserAgent is a better name

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,9 @@
+7.1.0
```
Comment (aelse) [inline CHANGELOG.md]: Agreed on 7.1 vs 8.0

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,9 @@
...
+-----
+- Datadog user-agent has changed to `gostatsd` by default.  Can now be configured.
+- New Datadog option: `user-agent`, configures the user agent supplied to Datadog.  Use `python-requests/2.6.0 CPython/2.7.10`
+  for old behavior.
```
Comment (aelse) [inline CHANGELOG.md]: I'm not really sold on the idea that this should be configurable. I can't see a use for this other than to masquerade as the datadog agent and then what we're really trying to do is subvert Datadog's view of which clients access it. I think this is something we could add in down the track if there are good reasons for it (eg. in the unlikely event that someone from Datadog asks us to). It's not likely that Datadog would attempt to block access to its customers by using the User-Agent, and they'd have to have a pretty good reason to do so.

File: pkg/backends/datadog/datadog.go (hunk 1 above)
Comment (tiedotguy) [reply pkg/backends/datadog/datadog.go]: Ahh, yeah.  I initially didn't change the value, only made it configurable.  Will update this.

File: CHANGELOG.md (hunk 3 above)
Comment (tiedotguy) [reply CHANGELOG.md]: I prefer to have it and not need it. It's an extra bit now vs another cycle later.

**Pull request merged**